src/components/Dialogs/DelegateDialog/DelegateDialog.tsx

```diff
@@ -57,16 +75,30 @@ export function DelegateDialog({
     address: delegatee?.delegatee as `0x${string}`,
   });
 
-  const { isError, writeContract: write, data } = useWriteContract();
+  const {
+    isError,
+    writeContract: write,
+    data: delegateTxHash,
+  } = useWriteContract();
 
   const {
     isLoading: isProcessingDelegation,
     isSuccess: didProcessDelegation,
     isError: didFailDelegation,
   } = useWaitForTransactionReceipt({
-    hash: data,
+    hash: ui.toggle("sponsoredDelegate") ? sponsoredVoteTxHash : delegateTxHash,
```

Comment:
Please create a separate dialog for gasless voting, perhaps named DelegateDialogGasless.tsx. This will make it easier for us to enable this component at the top level for specific tenants.